osu.Game/Screens/Play/Player.cs

```diff
@@ -225,6 +229,8 @@ private void load(AudioManager audio, OsuConfigManager config)
 
             DrawableRuleset.HasReplayLoaded.BindValueChanged(_ => updatePauseOnFocusLostState(), true);
 
+            confineMouseTracker?.OverlayActivationMode.BindTo(OverlayActivationMode);
```

Comment:
First of all, this is incorrect. You're binding here but never unbinding, relying on luck as to whether the original state revert is ever being sent to the tracker instance (depends on GC luck as bindable relationships are using weak references).

Why does this need to be done inside `Player`? The target bindable (`OverlayActivationMode`) is available in `OsuGame`, the same place you're making this tracker instance. I think binding at a game-wide level seems fine for now? Any case where we are disabling all overlays from displaying sounds like a case where we would be confining mouse input (should the user want that) anyways.

I'd also add xmldoc in the `OverlayActivation` enum class on a per-entry level, including mentioning the confine mode logic.